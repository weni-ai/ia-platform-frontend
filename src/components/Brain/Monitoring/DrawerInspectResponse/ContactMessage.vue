<template>
  <section class="inspect-response__contact">
    <UnnnicIntelligenceText
      color="neutral-cloudy"
      family="secondary"
      size="body-gt"
      tag="p"
    >
      {{ $t('router.monitoring.inspect_response.contact_message') }}:
    </UnnnicIntelligenceText>
    <p class="contact__text">“{{ text }}”</p>
  </section>
</template>

<script setup>
defineProps({
  text: {
    type: String,
    required: true,
  },
});
</script>
