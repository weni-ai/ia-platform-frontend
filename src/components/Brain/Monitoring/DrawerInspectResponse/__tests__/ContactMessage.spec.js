import { shallowMount } from '@vue/test-utils';
import { describe, it, expect, beforeEach } from 'vitest';

import ContactMessage from '../ContactMessage.vue';

describe('ContactMessage.vue', () => {
  let wrapper;
  const messageMock = 'Contact message mock';

  beforeEach(() => {
    wrapper = shallowMount(ContactMessage, {
      props: {
        text: messageMock,
      },
    });
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders the contact message title with correct props', () => {
    const contactMessageTitle = wrapper.findComponent(
      '[data-testid="contact-message-title"]',
    );
    expect(contactMessageTitle.exists()).toBe(true);
    expect(contactMessageTitle.props()).toMatchObject({
      color: 'neutral-cloudy',
      family: 'secondary',
      size: 'body-gt',
      tag: 'p',
    });
  });

  it('should renders the contact message text', () => {
    const contactMessageText = wrapper.find(
      '[data-testid="contact-message-text"]',
    );
    expect(contactMessageText.exists()).toBe(true);
    expect(contactMessageText.text()).toBe(`“${messageMock}”`);
  });

  it('should applies the correct class to the contact message container', () => {
    const contactMessageContainer = wrapper.find(
      '[data-testid="contact-message-container"]',
    );
    expect(contactMessageContainer.classes()).toContain(
      'inspect-response__contact',
    );
  });
});
