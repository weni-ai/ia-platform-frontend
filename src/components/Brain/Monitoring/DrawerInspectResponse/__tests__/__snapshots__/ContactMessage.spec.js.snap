// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactMessage.vue > matches the snapshot 1`] = `
"<section data-testid="contact-message-container" class="inspect-response__contact">
  <unnnic-intelligence-text-stub size="body-gt" tag="p" family="secondary" lineheight="md" weight="regular" color="neutral-cloudy" data-testid="contact-message-title"></unnnic-intelligence-text-stub>
  <p data-testid="contact-message-text" class="contact__text"> “Contact message mock” </p>
</section>"
`;
